Bottom: 523dfa57e04c205d1e8c691b23484644ed4eb4ac
Top:    a9f99a0aefe3876ca421579d10b6536c7c37b669
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-03-25 19:52:29 +0000

Add profiling to file load

















---

diff --git a/src/file.c b/src/file.c
index b6c67b9..018b272 100644
--- a/src/file.c
+++ b/src/file.c
@@ -350,6 +350,10 @@ LoadPCB (char *Filename)
 {
   PCBTypePtr newPCB = CreateNewPCB (False);
   Boolean units_mm;
+  clock_t start, end;
+  double elapsed;
+
+  start = clock();
 
   /* new data isn't added to the undo list */
   if (!ParsePCB (newPCB, Filename))
@@ -380,6 +384,10 @@ LoadPCB (char *Filename)
 	  CreateDefaultFont ();
 	}
 
+  end = clock();
+  elapsed = ((double) (end - start)) / CLOCKS_PER_SEC;
+  printf ("Loading file %s took %f\n", Filename, elapsed);
+
       /* clear 'changed flag' */
       SetChangedFlag (False);
       PCB->Filename = MyStrdup (Filename, "LoadPCB()");
